### PR TITLE
Add donate feature

### PR DESCRIPTION
**Pull request by haythem96**
# Task: Add donate feature

## High Level Description

Add `donate()` to send a specific asset amount to `MarginPool` while still getting accounted in `assetBalance`.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/Controller.sol [hunk 1]
```diff
@@ -260,6 +262,12 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
         _refreshConfigInternal();
     }
 
+    function donate(address _asset, uint256 _amount) external {
```
Comment (antoncoding) [inline contracts/Controller.sol:265]: Add comment to this! 😄

**Review by antoncoding — APPROVED** [review]

File: contracts/Controller.sol (hunk 1 above)
Comment (CruzMolina) [reply contracts/Controller.sol:265]: Second this 🙂.

**Review by aparnakr — APPROVED** [review]

**Comment by aparnakr** [comment]
What is the donate feature for? :D

**Review by CruzMolina — CHANGES_REQUESTED** [review]
https://github.com/opynfinance/GammaProtocol/pull/399/files#r625758655

Also curious about the idea/story for this.
Was the donate feature meant to be a part of #382 but left out?

**Review by antoncoding — APPROVED** [review]

**Pull request merged**